examples/gno.land/p/demo/forms/submit_test.gno

```diff
@@ -0,0 +1,95 @@
+package forms
+
+import (
+	"time"
+
+	"testing"
+)
+
+func TestAnswerForm(t *testing.T) {
+	db := NewDatabase()
+
+	data := `[
+		{
+			"label": "Name",
+			"fieldType": "string",
+			"required": true
+		},
+		{
+			"label": "Age",
+			"fieldType": "number",
+			"required": false
+		},
+		{
+			"label": "Is this a test?",
+			"fieldType": "boolean",
+			"required": false
+		},
+		{
+			"label": "Favorite Food",
+			"fieldType": "[Pizza|Schnitzel|Burger]",
+			"required": true
+		},
+		{
+			"label": "Favorite Foods",
+			"fieldType": "{Pizza|Schnitzel|Burger}",
+			"required": true
+		}
+	]`
+
+	formID := db.CreateForm("Test Form", "Test Description", "", "", data)
+	answers := `["Alex", 21, true, 0, [0, 1]]`
+
+	db.SubmitForm(formID, answers)
+
+	if len(db.Answers) != 1 {
+		t.Errorf("Expected 1 answer, got %d", len(db.Answers))
+	}
+
+	if db.Answers[0].FormID != formID {
+		t.Errorf("Expected form ID %s, got %s", formID, db.Answers[0].FormID)
+	}
+
+	if db.Answers[0].Answers != answers {
+		t.Errorf("Expected answers %s, got %s", answers, db.Answers[0].Answers)
+	}
+}
+
+func shouldPanic(t *testing.T, f func()) {
```

Comment:
Now that `p/demo/urequire` & `p/demo/uassert` you might want to use them for testing

Reply:
done in https://github.com/gnolang/gno/pull/2604/commits/8d97b6d871a495754bc8aaa279132c82b6e183ce